stl/inc/optional

```diff
@@ -104,7 +98,7 @@ struct _Optional_destruct_base<_Ty, false> { // either contains a value of _Ty o
 
     _CONSTEXPR20 ~_Optional_destruct_base() noexcept {
         if (_Has_value) {
-            _Destroy_in_place(_Value);
+            _Value.~decltype(_Value)();

```

Comment:
```suggestion
            _Value.~_Ty();
```
? There should not be a difference here even if `_Ty` is cv-qualified.

Reply:
Additionally, `~decltype` makes me immediately concerned about compiler bugs. (Weird syntax pattern-matches in my brain to "hmm, C1XX might not expect that".)